From f37e880d5ba77257f8ccd69016a2bcd71b382631 Mon Sep 17 00:00:00 2001
From: Thomas D.
Date: Tue, 7 Apr 2015 00:28:58 +0200
Subject: [PATCH 2/2] Don't use '-Werror'

---
 mklove/modules/configure.good_cflags | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mklove/modules/configure.good_cflags b/mklove/modules/configure.good_cflags
index 691e8b4..5fecfeb 100644
--- a/mklove/modules/configure.good_cflags
+++ b/mklove/modules/configure.good_cflags
@@ -9,5 +9,5 @@
 
 function checks {
     mkl_mkvar_append CPPFLAGS CPPFLAGS \
-        "-Wall -Werror -Wfloat-equal -Wpointer-arith"
+        "-Wall -Wfloat-equal -Wpointer-arith"
 }
-- 
2.3.5

