From fee964b4e9727087f31dd1d2bcc1514c929f8930 Mon Sep 17 00:00:00 2001
From: Thomas D.
Date: Tue, 7 Apr 2015 00:28:16 +0200
Subject: [PATCH 1/2] Don't add debug symbols per default

---
 mklove/modules/configure.cc | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/mklove/modules/configure.cc b/mklove/modules/configure.cc
index 3c24d80..631c003 100644
--- a/mklove/modules/configure.cc
+++ b/mklove/modules/configure.cc
@@ -79,21 +79,6 @@ function checks {
     [[ ! -z $LDFLAGS ]]  && mkl_mkvar_set "LDFLAGS" "LDFLAGS" "$LDFLAGS"
     [[ ! -z $ARFLAGS ]]  && mkl_mkvar_set "ARFLAGS" "ARFLAGS" "$ARFLAGS"
 
-
-    # Add debug symbol flag (-g)
-    # OSX 10.9 requires -gstrict-dwarf for some reason.
-    mkl_meta_set cc_g_dwarf name "debug symbols compiler flag (-g...)"
-    if [[ $MKL_DISTRO == "osx" ]]; then
-	if mkl_compile_check cc_g_dwarf "" cont CC "-gstrict-dwarf"; then
-	    mkl_mkvar_append CPPFLAGS CPPFLAGS "-gstrict-dwarf"
-	else
-	    mkl_mkvar_append CPPFLAGS CPPFLAGS "-g"
-	fi
-    else
-	mkl_mkvar_append CPPFLAGS CPPFLAGS "-g"
-    fi
-
-
     # pkg-config
     if [ -z "$PKG_CONFIG" ]; then
         PKG_CONFIG=pkg-config
-- 
2.3.5

