--- functions.sh.old	2014-05-09 06:21:14.993219139 +0200
+++ functions.sh	2014-05-09 06:34:54.302464884 +0200
@@ -438,9 +438,11 @@
 done
 
 # Setup COLS and ENDCOL so eend can line up the [ ok ]
-COLS="$(set -- $(stty size 2>/dev/null) ; printf "$2\n")"
+COLS="${COLUMNS:-0}"            # bash's internal COLUMNS variable
+[ "$COLS" -eq 0 ] && \
+        COLS="$(set -- $(stty size 2>/dev/null) ; printf "$2\n")"
 [ -z "$COLS" ] && COLS=80
-[ "$COLS" -gt 0 ] || COLS=80	# width of [ ok ] == 7
+[ "$COLS" -gt 0 ] || COLS=80    # width of [ ok ] == 7
 
 if yesno "${RC_ENDCOL}"; then
 	ENDCOL='\033[A\033['$(( COLS - 8 ))'C'
