--- udev-187/src/udev/udev-event.c
+++ udev-187/src/udev/udev-event.c
@@ -689,6 +689,8 @@ int udev_event_spawn(struct udev_event *event,
         /* allow programs in /usr/lib/udev/ to be called without the path */
         if (argv[0][0] != '/') {
                 util_strscpyl(program, sizeof(program), UDEVLIBEXECDIR "/", argv[0], NULL);
+                if(access(program, X_OK))
+                        util_strscpyl(program, sizeof(program), "/usr/lib/udev/", argv[0], NULL);
                 argv[0] = program;
         }
 
