--- install.sh.old	2013-09-09 16:51:03.892699898 +0200
+++ install.sh	2013-09-09 16:52:24.959281632 +0200
@@ -186,46 +186,52 @@
 INSTALLD='-D'
 
 if [ -z "$BUILD" ]; then
-    case $(uname) in
-	cygwin*)
-	    BUILD=cygwin
-	    ;;
-	Darwin)
-	    BUILD=apple
-	    ;;
-	*)
-	    if [ -f /etc/os-release ]; then
-		eval $(cat /etc/os-release | grep ^ID)
-
-		case $ID in
-		    fedora)
-			BUILD=redhat
-			;;
-		    debian)
-			BUILD=debian
+	case $(uname) in
+		cygwin*)
+			BUILD=cygwin
 			;;
-		    opensuse)
-			BUILD=suse
+		Darwin)
+			BUILD=apple
 			;;
-		    *)
-			BUILD="$ID"
+		*)
+			if [ -f /etc/os-release ]; then
+				eval $(cat /etc/os-release | grep ^ID)
+				
+				case $ID in
+					debian)
+						BUILD=debian
+						;;
+					fedora)
+						BUILD=redhat
+						;;
+					gentoo)
+						BUILD=gentoo
+						;;
+					opensuse)
+						BUILD=suse
+						;;
+					*)
+						BUILD="$ID"
+						;;
+				esac
+			
+			elif [ -f /etc/arch-release ]; then
+				BUILD=archlinux
+			elif [ -f /etc/debian_version ]; then
+				BUILD=debian
+			elif [ -f /etc/gentoo-release ]; then
+				BUILD=gentoo
+			elif [ -f /etc/redhat-release ]; then
+				BUILD=redhat
+			elif [ -f /etc/slackware-version ]; then
+				BUILD=slackware
+			elif [ -f /etc/SuSE-release ]; then
+				BUILD=suse
+			else
+				BUILD=linux
+			fi
 			;;
-		esac
-	    elif [ -f /etc/debian_version ]; then
-		BUILD=debian
-	    elif [ -f /etc/redhat-release ]; then
-		BUILD=redhat
-	    elif [ -f /etc/slackware-version ] ; then
-		BUILD=slackware
-	    elif [ -f /etc/SuSE-release ]; then
-		BUILD=suse
-	    elif [ -f /etc/arch-release ] ; then
-		BUILD=archlinux
-	    else
-		BUILD=linux
-	    fi
-	    ;;
-    esac
+	esac
 fi
 
 case $BUILD in
@@ -265,18 +271,18 @@
 [ -n "$HOST" ] || HOST=$BUILD
 
 case "$HOST" in
-    cygwin)
-	echo "Installing Cygwin-specific configuration..."
-	;;
-    apple)
-	echo "Installing Mac-specific configuration...";
-	;;
-    debian|redhat|slackware|archlinux|linux|suse)
-	;;
-    *)
-	echo "ERROR: Unknown HOST \"$HOST\"" >&2
-	exit 1;
-	;;
+	apple)
+		echo "Installing Mac-specific configuration...";
+		;;
+	cygwin)
+		echo "Installing Cygwin-specific configuration..."
+		;;
+	archlinux|debian|gentoo|linux|redhat|slackware|suse)
+		;;
+	*)
+		echo "ERROR: Unknown HOST \"$HOST\"" >&2
+		exit 1;
+		;;
 esac
 
 if [ -z "$file" ]; then
