--- install.sh.old	2013-09-09 16:12:59.510214756 +0200
+++ install.sh	2013-09-09 16:15:45.017389362 +0200
@@ -194,46 +194,52 @@
 T='-T'
 
 if [ -z "$BUILD" ]; then
-    case $(uname) in
-	cygwin*)
-	    BUILD=cygwin
-	    ;;
-	Darwin)
-	    BUILD=apple
-	    ;;
-	*)
-	    if [ -f /etc/os-release ]; then
-		eval $(cat /etc/os-release | grep ^ID)
-
-		case $ID in
-		    fedora)
-			BUILD=redhat
-			;;
-		    debian)
-			BUILD=debian
+	case $(uname) in
+		cygwin*)
+			BUILD=cygwin
 			;;
-		    opensuse)
-			BUILD=suse
+		Darwin)
+			BUILD=apple
 			;;
-		    *)
-			BUILD="$ID"
+		*)
+			if [ -f /etc/os-release ]; then
+				eval $(cat /etc/os-release | grep ^ID)
+				
+				case $ID in
+					debian)
+						BUILD=debian
+						;;
+					fedora)
+						BUILD=redhat
+						;;
+					gentoo)
+						BUILD=gentoo
+						;;
+					opensuse)
+						BUILD=suse
+						;;
+					*)
+						BUILD="$ID"
+						;;
+				esac
+			
+			elif [ -f /etc/arch-release ]; then
+				BUILD=archlinux
+			elif [ -f /etc/debian_version ]; then
+				BUILD=debian
+			elif [ -f /etc/gentoo-release ]; then
+				BUILD=gentoo
+			elif [ -f /etc/redhat-release ]; then
+				BUILD=redhat
+			elif [ -f /etc/slackware-version ]; then
+				BUILD=slackware
+			elif [ -f /etc/SuSE-release ]; then
+				BUILD=suse
+			else
+				BUILD=linux
+			fi
 			;;
-		esac
-	    elif [ -f ${CONFDIR}/debian_version ]; then
-		BUILD=debian
-	    elif [ -f ${CONFDIR}/redhat-release ]; then
-		BUILD=redhat
-	    elif [ -f ${CONFDIR}/SuSE-release ]; then
-		BUILD=suse
-	    elif [ -f ${CONFDIR}/slackware-version ] ; then
-		BUILD=slackware
-	    elif [ -f ${CONFDIR}/arch-release ] ; then
-		BUILD=archlinux
-	    else
-		BUILD=linux
-	    fi
-	    ;;
-    esac
+	esac
 fi
 
 case $BUILD in
@@ -258,35 +264,38 @@
 [ -n "$HOST" ] || HOST=$BUILD
 
 case "$HOST" in
-    cygwin)
-	echo "$PRODUCT is not supported on Cygwin" >&2
-	exit 1
-	;;
-    apple)
-	echo "$PRODUCT is not supported on OS X" >&2
-	exit 1
-	;;
-    debian)
-	echo "Installing Debian-specific configuration..."
-	;;
-    redhat)
-	echo "Installing Redhat/Fedora-specific configuration..."
-	;;
-    slackware)
-	echo "Installing Slackware-specific configuration..."
-	;;
-    archlinux)
-	echo "Installing ArchLinux-specific configuration..."
-	;;
-    suse)
-	echo "Installing Suse-specific configuration..."
-	;;
-    linux)
-	;;
-    *)
-	echo "ERROR: Unknown HOST \"$HOST\"" >&2
-	exit 1;
-	;;
+	apple)
+		echo "$PRODUCT is not supported on OS X" >&2
+		exit 1
+		;;
+	archlinux)
+		echo "Installing ArchLinux-specific configuration..."
+		;;
+	cygwin)
+		echo "$PRODUCT is not supported on Cygwin" >&2
+		exit 1
+		;;
+	debian)
+		echo "Installing Debian-specific configuration..."
+		;;
+	gentoo)
+		echo "Installing Gentoo-specific configuration..."
+		;;
+	redhat)
+		echo "Installing Redhat/Fedora-specific configuration..."
+		;;
+	slackware)
+		echo "Installing Slackware-specific configuration..."
+		;;
+	suse)
+		echo "Installing Suse-specific configuration..."
+		;;
+	linux)
+		;;
+	*)
+		echo "ERROR: Unknown HOST \"$HOST\"" >&2
+		exit 1;
+		;;
 esac
 
 [ -z "$INITDIR" ] && INITDIR="${CONFDIR}/init.d"
@@ -388,7 +397,10 @@
 fi
 
 if [ $HOST = archlinux ] ; then
-   sed -e 's!LOGFILE=/var/log/messages!LOGFILE=/var/log/messages.log!' -i ${DESTDIR}${CONFDIR}/$PRODUCT/$PRODUCT.conf
+	sed -e 's!LOGFILE=/var/log/messages!LOGFILE=/var/log/messages.log!' -i ${DESTDIR}${CONFDIR}/$PRODUCT/$PRODUCT.conf
+elif [ $HOST = gentoo ]; then
+	# Adjust SUBSYSLOCK path (see https://bugs.gentoo.org/show_bug.cgi?id=459316)
+	perl -p -w -i -e "s|^SUBSYSLOCK=.*|SUBSYSLOCK=/run/lock/$PRODUCT|;" ${DESTDIR}${CONFDIR}/$PRODUCT/$PRODUCT.conf
 fi
 
 #
