Add missing files to run test-suite

Bug: https://github.com/rsyslog/rsyslog/pull/707

diff -rupN old/rsyslog-8.15.0/tests/json_null_array.sh new/rsyslog-8.15.0/tests/json_null_array.sh
--- old/rsyslog-8.15.0/tests/json_null_array.sh	1970-01-01 01:00:00.000000000 +0100
+++ new/rsyslog-8.15.0/tests/json_null_array.sh	2015-12-16 19:54:13.000000000 +0100
@@ -0,0 +1,14 @@
+#!/bin/bash
+# added 2015-11-17 by rgerhards
+# This file is part of the rsyslog project, released under ASL 2.0
+echo ===============================================================================
+echo \[json_null_array.sh\]: test for json containung \"null\" value
+. $srcdir/diag.sh init
+. $srcdir/diag.sh startup json_null_array.conf
+. $srcdir/diag.sh tcpflood -m 1 -M "\"<167>Mar  6 16:57:54 172.20.245.8 test: @cee: { \\\"array\\\": [0, 1, null, 2, 3, null, 4] }\""
+echo doing shutdown
+. $srcdir/diag.sh shutdown-when-empty
+echo wait on shutdown
+. $srcdir/diag.sh wait-shutdown
+. $srcdir/diag.sh seq-check 0 4
+. $srcdir/diag.sh exit
diff -rupN old/rsyslog-8.15.0/tests/json_null_array-vg.sh new/rsyslog-8.15.0/tests/json_null_array-vg.sh
--- old/rsyslog-8.15.0/tests/json_null_array-vg.sh	1970-01-01 01:00:00.000000000 +0100
+++ new/rsyslog-8.15.0/tests/json_null_array-vg.sh	2015-12-16 19:54:13.000000000 +0100
@@ -0,0 +1,14 @@
+#!/bin/bash
+# added 2015-11-17 by rgerhards
+# This file is part of the rsyslog project, released under ASL 2.0
+echo ===============================================================================
+echo \[json_null_array.sh\]: test for json containung \"null\" value
+. $srcdir/diag.sh init
+. $srcdir/diag.sh startup-vg json_null_array.conf
+. $srcdir/diag.sh tcpflood -m 1 -M "\"<167>Mar  6 16:57:54 172.20.245.8 test: @cee: { \\\"array\\\": [0, 1, null, 2, 3, null, 4] }\""
+echo doing shutdown
+. $srcdir/diag.sh shutdown-when-empty
+echo wait on shutdown
+. $srcdir/diag.sh wait-shutdown-vg
+. $srcdir/diag.sh seq-check 0 4
+. $srcdir/diag.sh exit
diff -rupN old/rsyslog-8.15.0/tests/json_null.sh new/rsyslog-8.15.0/tests/json_null.sh
--- old/rsyslog-8.15.0/tests/json_null.sh	1970-01-01 01:00:00.000000000 +0100
+++ new/rsyslog-8.15.0/tests/json_null.sh	2015-12-16 19:54:13.000000000 +0100
@@ -0,0 +1,15 @@
+#!/bin/bash
+# added 2015-11-17 by rgerhards
+# This file is part of the rsyslog project, released under ASL 2.0
+# Note: the aim of this test is to test against misadressing, so we do
+# not actually check the output
+echo ===============================================================================
+echo \[json_null.sh\]: test for json containung \"null\" value
+. $srcdir/diag.sh init
+. $srcdir/diag.sh startup json_null.conf
+. $srcdir/diag.sh tcpflood -m 1 -M "\"<167>Mar  6 16:57:54 172.20.245.8 test: @cee: { \\\"nope\\\": null }\""
+echo doing shutdown
+. $srcdir/diag.sh shutdown-when-empty
+echo wait on shutdown
+. $srcdir/diag.sh wait-shutdown
+. $srcdir/diag.sh exit
diff -rupN old/rsyslog-8.15.0/tests/json_null-vg.sh new/rsyslog-8.15.0/tests/json_null-vg.sh
--- old/rsyslog-8.15.0/tests/json_null-vg.sh	1970-01-01 01:00:00.000000000 +0100
+++ new/rsyslog-8.15.0/tests/json_null-vg.sh	2015-12-16 19:54:13.000000000 +0100
@@ -0,0 +1,15 @@
+#!/bin/bash
+# added 2015-11-17 by rgerhards
+# This file is part of the rsyslog project, released under ASL 2.0
+# Note: the aim of this test is to test against misadressing, so we do
+# not actually check the output
+echo ===============================================================================
+echo \[json_null.sh\]: test for json containung \"null\" value
+. $srcdir/diag.sh init
+. $srcdir/diag.sh startup-vg json_null.conf
+. $srcdir/diag.sh tcpflood -m 1 -M "\"<167>Mar  6 16:57:54 172.20.245.8 test: @cee: { \\\"nope\\\": null }\""
+echo doing shutdown
+. $srcdir/diag.sh shutdown-when-empty
+echo wait on shutdown
+. $srcdir/diag.sh wait-shutdown-vg
+. $srcdir/diag.sh exit
diff -rupN old/rsyslog-8.15.0/tests/Makefile.am new/rsyslog-8.15.0/tests/Makefile.am
--- old/rsyslog-8.15.0/tests/Makefile.am	2015-12-15 07:51:54.000000000 +0100
+++ new/rsyslog-8.15.0/tests/Makefile.am	2015-12-16 20:48:04.000000000 +0100
@@ -827,7 +827,7 @@ EXTRA_DIST= \
 	mysql-actq-mt-withpause-vg.sh \
 	testsuites/mysql-actq-mt.conf \
 	mmpstrucdata.sh \
-        mmpstrucdata-vg.sh \
+	mmpstrucdata-vg.sh \
 	testsuites/mmpstrucdata.conf \
 	mmpstrucdata-invalid-vg.sh \
 	testsuites/mmpstrucdata-invalid.conf \
@@ -861,6 +861,12 @@ EXTRA_DIST= \
 	json_array_looping.sh \
 	json_nonarray_looping.sh \
 	testsuites/json_array_looping.conf \
+	json_null.sh \
+	json_null-vg.sh \
+	testsuites/json_null.conf \
+	json_null_array.sh \
+	json_null_array-vg.sh \
+	testsuites/json_null_array.conf \
 	mmnormalize_regex.sh \
 	testsuites/mmnormalize_regex.conf \
 	testsuites/mmnormalize_regex.rulebase \
diff -rupN old/rsyslog-8.15.0/tests/testsuites/json_null_array.conf new/rsyslog-8.15.0/tests/testsuites/json_null_array.conf
--- old/rsyslog-8.15.0/tests/testsuites/json_null_array.conf	1970-01-01 01:00:00.000000000 +0100
+++ new/rsyslog-8.15.0/tests/testsuites/json_null_array.conf	2015-12-16 19:54:13.000000000 +0100
@@ -0,0 +1,12 @@
+$IncludeConfig diag-common.conf
+module(load="../plugins/mmjsonparse/.libs/mmjsonparse")
+module(load="../plugins/imtcp/.libs/imtcp")
+input(type="imtcp" port="13514")
+
+template(name="outfmt" type="string" string="%$.data%\n")
+
+action(type="mmjsonparse")
+foreach ($.data in $!array) do {
+	if not ($.data == "") then
+		action(type="omfile" file="rsyslog.out.log" template="outfmt")
+}
diff -rupN old/rsyslog-8.15.0/tests/testsuites/json_null.conf new/rsyslog-8.15.0/tests/testsuites/json_null.conf
--- old/rsyslog-8.15.0/tests/testsuites/json_null.conf	1970-01-01 01:00:00.000000000 +0100
+++ new/rsyslog-8.15.0/tests/testsuites/json_null.conf	2015-12-16 19:54:13.000000000 +0100
@@ -0,0 +1,14 @@
+$IncludeConfig diag-common.conf
+module(load="../plugins/mmjsonparse/.libs/mmjsonparse")
+module(load="../plugins/imtcp/.libs/imtcp")
+input(type="imtcp" port="13514")
+
+# we must make sure the template contains a reference to the 
+# data item with null value
+template(name="outfmt" type="string" string="%$!nope%\n")
+template(name="outfmt-all-json" type="string" string="%$!all-json%\n")
+
+action(type="mmjsonparse")
+action(type="omfile" file="./rsyslog.out.log" template="outfmt")
+if $!nope == "" then
+	action(type="omfile" file="./rsyslog2.out.log" template="outfmt-all-json")
