diff -rupN old/salt-2014.7.1/tests/unit/pydsl_test.py new/salt-2014.7.1/tests/unit/pydsl_test.py
--- old/salt-2014.7.1/tests/unit/pydsl_test.py	2015-01-14 19:38:22.000000000 +0100
+++ new/salt-2014.7.1/tests/unit/pydsl_test.py	2015-02-20 15:18:09.631430680 +0100
@@ -372,6 +372,13 @@ class PyDSLRendererTestCase(TestCase):
                     state(color).cmd.run('echo hello '+color+' '+str(number)+' >> {3}', cwd='/')
                 '''.format(output, output, output, output)))
 
+            # Because salt.fileserver caches for 30sec per default
+            # we need to make sure to wait at least 30sec before calling
+            # the highstate or salt won't find the new created files
+            # (see upstream issue #20504 for more details)
+            import time
+            time.sleep(30)
+
             state_highstate({'base': ['aaa']}, dirpath)
             expected = textwrap.dedent('''\
                 X1
